src/controllers/Chef.ts

```diff
@@ -1,21 +1,78 @@
 import { NextFunction, Request, Response } from 'express';
-import mongoose, { Mongoose } from 'mongoose';
+import mongoose from 'mongoose';
 import Chef from '../models/Chef';
 
-const createChef = (req: Request, res: Response, next: NextFunction) => {
-    const { name } = req.body;
+const createChef = async (req: Request, res: Response, next: NextFunction) => {
+    const { name, image, info } = req.body;
     const chef = new Chef({
         _id: new mongoose.Types.ObjectId(),
-        name
+        name,
+        image,
+        info
     });
-    return chef
-        .save()
-        .then((chef) => res.status(201).json({ chef }))
-        .catch((error) => {
-            res.status(500).json({ error });
-        });
+
+    try {
+        const savedChef = await chef.save();
+        res.status(201).json({ chef: savedChef });
+    } catch (error) {
+        res.status(500).json({ error });
+    }
+};
+
+const readChef = async (req: Request, res: Response, next: NextFunction) => {
+    const chefId = req.params.chefId;
+
+    try {
+        const chef = await Chef.findById(chefId);
+        if (chef) {
+            res.status(200).json({ chef });
+        } else {
+            res.status(404).json({ message: 'Not found' });
+        }
+    } catch (error) {
+        res.status(500).json({ error });
+    }
 };
-const readChef = (req: Request, res: Response, next: NextFunction) => {};
-const readAllChef = (req: Request, res: Response, next: NextFunction) => {};
-const updateChef = (req: Request, res: Response, next: NextFunction) => {};
-const deleteChef = (req: Request, res: Response, next: NextFunction) => {};
+
+const readAllChef = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const chefs = await Chef.find();
+        res.status(200).json({ chefs });
+    } catch (error) {
+        res.status(500).json({ error });
+    }
+};
+
+const updateChef = async (req: Request, res: Response, next: NextFunction) => {
+    const chefId = req.params.chefId;
+
+    try {
+        const chef = await Chef.findById(chefId);
+        if (chef) {
+            chef.set(req.body);
```

Comment:
be careful setting the whole req.body into the database, it would be better to either destructure the chef properties one by one, or receive it over a chef key from the beginning